enclave/core/backtrace.c

```diff
@@ -2,46 +2,15 @@
 // Licensed under the MIT License.
 
 #include <openenclave/bits/safecrt.h>
+#include <openenclave/corelibc/stdio.h>
 #include <openenclave/enclave.h>
 #include <openenclave/internal/backtrace.h>
 #include <openenclave/internal/calls.h>
 #include <openenclave/internal/globals.h>
+#include <openenclave/internal/malloc.h>
 #include <openenclave/internal/print.h>
 #include <openenclave/internal/raise.h>
 
-char** oe_backtrace_symbols(void* const* buffer, int size)
-{
-    char** ret = NULL;
-    oe_backtrace_symbols_args_t* args = NULL;
-
-    if (!buffer || size > OE_BACKTRACE_MAX)
-        goto done;
-
-    if (!(args = oe_host_malloc(sizeof(oe_backtrace_symbols_args_t))))
-        goto done;
-
-    if (oe_memcpy_s(
-            args->buffer,
-            sizeof(void*) * OE_BACKTRACE_MAX,
-            buffer,
-            sizeof(void*) * (size_t)size) != OE_OK)
-        goto done;
-    args->size = size;
-    args->ret = NULL;
-
-    if (oe_ocall(OE_OCALL_BACKTRACE_SYMBOLS, (uint64_t)args, NULL) != OE_OK)
-        goto done;
-
-    ret = args->ret;
-
-done:
-
-    if (args)
-        oe_host_free(args);
-
-    return ret;
-}
-
 oe_result_t oe_print_backtrace(void)
```

Comment:
I added issue #2048 because I believe **oe_assert()** should call this function before aborting.

Reply:
Also, the static size is necessary when calling the Linux **backtrace()** function whose signature is taken verbatim for **oe_backtrace()**. Avoiding the static buffer would require some rework and would be a deviation from the Linux signature. Also, since this is preexisting, I'd rather treat it separately from thei PR.